File: solutions/java/armstrong-numbers/1/src/main/java/ArmstrongNumbers.java
import java.util.*;

class ArmstrongNumbers {
    
    boolean isArmstrongNumber(int numberToCheck) {
        if (numberToCheck < 10) {
            return true;
        }

        ArrayList digits = this.getDigits(numberToCheck);

        return this.armstrongSum(digits) == numberToCheck;
    }

    int armstrongSum(ArrayList digits) {
        int sum = 0;
        int exp = digits.size();
        Iterator d = digits.iterator();
        
        while (d.hasNext()) {
            int digit = (int) d.next();
            sum += Math.pow(digit, exp);
        }
        return sum;
    }

    ArrayList getDigits(int number) {
        ArrayList digits = new ArrayList();
        int copy = number;

        while (copy > 0) {
            digits.add(copy % 10);
            copy /= 10;
        }
        
        return digits;
    }

}

File: solutions/c/armstrong-numbers/1/armstrong_numbers.c
#include <math.h>
#include "armstrong_numbers.h"
bool is_armstrong_number(int candidate) {
    int backup = candidate;
    int length = 0;
    int sum = 0;
    int digit;

    while (candidate > 0) {
        length++;
        candidate /= 10;
    }

    candidate = backup;
    while (candidate > 0) {
        digit = candidate % 10;
        sum += pow(digit, length);
        candidate /= 10;
    }

    return backup == sum;
}